sky/skylet/job_lib.py

```diff
@@ -702,23 +703,37 @@ def load_job_queue(payload: str) -> List[Dict[str, Any]]:
     return jobs
 
 
-def cancel_jobs(job_owner: str, jobs: Optional[List[int]]) -> None:
-    """Cancel the jobs.
+def cancel_jobs_encoded_results(job_owner: str,
+                                jobs: Optional[List[int]],
+                                cancel_all: bool = False) -> str:
+    """Cancel jobs.
 
     Args:
-        jobs: The job ids to cancel. If None, cancel all the jobs.
+        jobs: Job IDs to cancel. If None, cancel the latest running job.
+        all: Whether to cancel all the jobs. If
+            set to True, asserts `jobs` is set to None.
+
+    Returns:
+        Encoded job IDs that are actually cancelled. Caller should use
+        common_utils.decode_payload() to parse.
     """
-    # Update the status of the jobs to avoid setting the status of stale
-    # jobs to CANCELLED.
-    if jobs is None:
+    if cancel_all:
+        # Cancel the latest (largest job ID) running job.
```

Comment:
comment needs update (should move to L727)

Reply:
Oops, good catch! Moved. Thanks!